prince-of-versions/src/test/java/co/infinum/princeofversions/tests/NetworkLoaderTest.java

```diff
@@ -0,0 +1,176 @@
+import android.os.Build;
+import android.util.Log;
+
+import java.io.IOException;
+
+import co.infinum.princeofversions.BuildConfig;
+import co.infinum.princeofversions.UpdateConfigLoader;
+import co.infinum.princeofversions.callbacks.UpdaterCallback;
+import co.infinum.princeofversions.common.ErrorCode;
+import co.infinum.princeofversions.common.VersionContext;
+import co.infinum.princeofversions.helpers.ContextHelper;
+import co.infinum.princeofversions.helpers.parsers.JsonVersionConfigParser;
+import co.infinum.princeofversions.interfaces.VersionRepository;
+import co.infinum.princeofversions.interfaces.VersionVerifier;
+import co.infinum.princeofversions.interfaces.VersionVerifierFactory;
+import co.infinum.princeofversions.interfaces.VersionVerifierListener;
+import co.infinum.princeofversions.loaders.NetworkLoader;
+import co.infinum.princeofversions.loaders.factories.NetworkLoaderFactory;
+import co.infinum.princeofversions.threading.ExecutorServiceVersionVerifier;
+import co.infinum.princeofversions.util.ResourceUtils;
+import co.infinum.princeofversions.verifiers.SingleThreadVersionVerifier;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Created by Juraj Pejnović on 07/09/16.
+ */
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = Build.VERSION_CODES.LOLLIPOP)
+public class NetworkLoaderTest {
+
+    private MockWebServer mockWebServer;
+
+    private VersionVerifier versionVerifier;
+
+    private VersionVerifierFactory provider;
+
+    private UpdaterCallback callback;
+
+    private VersionRepository repository;
+
+
+    private Context setupContext(String versionName) throws PackageManager.NameNotFoundException {
+        Context context = Mockito.mock(Context.class, Mockito.RETURNS_DEEP_STUBS);
+        Mockito.when(context.getPackageName()).thenReturn("name");
+        PackageInfo packageInfo = Mockito.mock(PackageInfo.class);
+        packageInfo.versionName = versionName;
+        Mockito.when(context.getPackageManager().getPackageInfo("name", 0)).thenReturn(packageInfo);
+        versionVerifier = new ExecutorServiceVersionVerifier(new JsonVersionConfigParser(ContextHelper.getAppVersion(context)));
+        Mockito.when(provider.newInstance()).thenReturn(versionVerifier);
+        return context;
+    }
+
+    @Before
+    public void setup() {
+        callback = Mockito.mock(UpdaterCallback.class);
+        provider = Mockito.mock(VersionVerifierFactory.class);
+
+        ShadowLog.stream = System.out;
+        repository = Mockito.mock(VersionRepository.class);
+        Mockito.when(repository.getLastVersionName(Mockito.anyString())).thenReturn(null);
+        Mockito.doNothing().when(repository).setLastVersionName(Mockito.anyString());
+
+        mockWebServer = new MockWebServer();
```

Comment:
You can start the server in this method. That way you do not have start it for every test.